.github/workflows/build.yml

```diff
@@ -0,0 +1,120 @@
+name: Compilation
+on:
+  push:
+
+jobs:
+  build:
+    name: Build
+    runs-on: ubuntu-latest
+    env:
+      REGISTRY: ghcr.io
+      IMAGE_NAME: ${{ github.repository }}
+    permissions: 
+      contents: read
+      packages: write 
+
+    steps:
+
+    - name: Set up Go 1.16
+      uses: actions/setup-go@v1
```

Comment:
Shall we pick a more recent (major) version (v2) of the setup-go action?
https://github.com/actions/setup-go/releases